### DISCUSSION TITLE
tud_cdc_write_available always returns 0

### DISCUSSION THREAD

**Discussion by caedmon** [hathach/tinyusb#1596] [Q&A]
I'm using tinusb on a RP2040 and everything works fine until I reboot at which point tud_cdc_write_available begins return 0 all the time. tud_cdc_write appears to still function normally.

It seems as though something doesn't properly reinitialize after the reboot. Power cycling gets it working again.

Is there some way to force a complete reinitialization?

**Comment by hathach** [comment]
`until I reboot` I guess you mean you reboot your PC right ? If it is the case, would you mind provide info of your PC e.g laptop/desktop and OS version. Also which firmware you are running, and how do you test tud_cdc_write_available() i.e exact test setup to reproduce the issue for troubleshooting.

**Reply by caedmon** [reply]
No, I mean reboot the RP2040, either using the reset/run pin or by using the watchdog timer. The firmware is custom, using pico-sdk and FreeRTOS.

**Reply by hathach** [reply]
can you provide your simplest/minimal code (build-able and run-able) and steps to reproduce the issue e.g run the code, press reset then observe which an out (led/serial) that illustrate the issue.

**Comment by caedmon** [comment]
The solution I found is to check tud_cdc_connected() first and only when it returns true do I check tud_cdc_write_available(). This seems to work both before and after reboot.

I have no idea why it doesn't appear to be necessary before a reboot.
